addons/block_code/ui/blocks/control_block/control_block.gd

```diff
@@ -83,7 +83,11 @@ func get_serialized_props() -> Array:
 	return props
 
 
-func get_scene_path():
+func get_class():
+	return "ControlBlock"
```

Comment:
Is there really no built-in way to ask a value for its class name?

Should these return `StringName` rather than `String`?

Reply:
OK, I see that the built-in way is … `get_class()` which returns `String`, and ignores `class_name`.

https://docs.godotengine.org/en/4.2/classes/class_object.html#class-object-method-get-class

Reply:
I will try with StringName and see how it goes!

As per the actual question, I couldn't find any for custom classes. A lot of discussion in issues like https://github.com/godotengine/godot/issues/21789

Reply:
I think String is correct – you're overriding a method that returns `String`.

I have tried some experiments and I couldn't find a better way either. Sad.

Reply:
@wjt it worked with StringName.

Reply:
I think you should keep it as a String because https://docs.godotengine.org/en/stable/classes/class_object.html#class-object-method-get-class has return type String. But maybe StringName is close enough to being a String…

Reply:
Either that, or give it a different name. The advantage of defining a new method is that when we inevitably forget to implement it for some new block type, we'll get an error rather than falling back to the built in `Object.get_class`

Reply:
@wjt yes good point, let's go with a new method to also avoid confusion.

Reply:
The other nice thing of using a separate method: we can mark it as static.